sp/src/game/server/episodic/grenade_hopwire.cpp

```diff
@@ -1137,10 +1147,22 @@ bool CGravityVortexController::TryCreateRecipeNPC( const char *szClass, const ch
 		baseNPC->AddSpawnFlags( SF_NPC_FALL_TO_GROUND );
 		baseNPC->m_tEzVariant = CAI_BaseNPC::EZ_VARIANT_XEN;
 
-		// Set the squad name of a new Xen NPC to 'xenpc_headcrab', 'xenpc_bullsquid', etc
-		char * squadname = UTIL_VarArgs( "xe%s", szClass );
-		DevMsg( "Adding xenpc '%s' to squad '%s' \n", baseNPC->GetDebugName(), squadname );
-		baseNPC->SetSquadName( AllocPooledString( squadname ) );
+
+		const char * squadnameContext = GetContextValue( "squadname" );
+
+		// If the context "squadname" does not match an empty string, use that instead of the default Xen squad name
+		if (!Matcher_Match( squadnameContext, "default" ))
```

Comment:
Why is `Matcher_Match` being used here instead of a more direct string comparison like `FStrEq` or `Q_stricmp`? The first parameter is also supposed to be the "query" parameter (i.e. the part that has wildcards and `<`, `&`, etc.)

Matchers should not be used for simple string comparisons. `Matcher_Match` is best used when a query can contain wildcards or the aforementioned comparison operators, which is useful for cases like numeric keyvalues inputted by a mapper. When comparison operators are not involved, `Matcher_NamesMatch` should be used instead, as it only does a string comparison with wildcards. However, I don't see a use for either of those in this case.

Is there a specific reason for this which I'm not understanding?

Reply:
This is a good call out. There is no good reason. I should be using FStrEq; I believe that was the method I was looking for.